src/CoreAnimation/CALayer.cs

```diff
@@ -103,6 +103,8 @@ void SetCALayerDelegate (CALayerDelegate? value)
 			}
 		}
 
+		// Note: preserving this member allows us to re-enable the `Optimizable` binding flag
```

Comment:
Why would preserving the OnDispose method make the calling method optimizable? Or am I missing something?

Reply:
The (dotnet) processing for (`Optimizable`)`Dispose` method body is skipped to avoid marking the backing fields.

When the bodies are re-introduced it's too late [1] to mark their content again. That works well for purely generated `Dispose` methods (since the `base.Dispose` is part of `IDisposable` it's already marked) and we check if the fields are marked (or not).

However if code (like calls to other methods) are injected then they might not be marked - so those extra pieces (like `OnDispose` here) have to be preserved (or the linker won't be able to save the _invalid_ assembly).

[1] there's something else I want to try... but it's more complex/time consuming than fixing the 3 usages we currently require